container/broker/lxd-broker.go

```diff
@@ -156,6 +160,37 @@ func (broker *lxdBroker) StartInstance(ctx context.ProviderCallContext, args env
 	}, nil
 }
 
+func selectBridgeDevice(agentCfg agent.Config) (string, error) {
+	// If the operator has already selected a bridge device use that
+	preferredDev := agentCfg.Value(agent.LxdBridge)
+	if preferredDev == "" {
+		preferredDev = agentCfg.Value(agent.LxcBridge)
+	}
+	if preferredDev != "" {
+		return preferredDev, nil
+	}
+
+	// Check if the device list contains any OVS-managed bridges. If only
+	// one OVS bridge is available prefer that; otherwise, log a warning
+	// and fall back to the default LXD bridge
+	ovsDeviceNames, err := lookupOvsManagedBridges()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	switch ovsDeviceNames.Size() {
+	case 0: // no OVS-bridges present
+		return jujunetwork.DefaultLXDBridge, nil
+	case 1:
+		preferredDev = ovsDeviceNames.Values()[0]
+		logger.Infof("using OVS-managed bridge device %q for containerized workloads", preferredDev)
+		return preferredDev, nil
+	default:
```

Comment:
I wonder whether we can automatically resolve this ambiguity by making the default block space-aware. Perhaps this will help us filter the list to a single OVS bridge that we can use based on space requirements provided by the operator.